src/services/archive.js

```diff
@@ -0,0 +1,223 @@
+  }
+
+  async getPlatformJson() {
+    const platformJsonPath = await this.getPlatformJsonPath();
+    const platformJsonFile = fs.readFileSync(platformJsonPath);
+    const jsonContent = JSON.parse(platformJsonFile);
+
+    return jsonContent;
+  }
+
+  async getArchiveData() {
+    if (!this.isInitialized()) {
+      const temporaryDir = await tmp.dir();
+
+      const destinationDir = temporaryDir.path;
+      const platformFileName = `platform.${ARCHIVE_FORMAT}`;
+
+      await downloadFileFollowRedirect(this.url, { directory: destinationDir, filename: platformFileName });
+
+      this.archiveData = {
+        path: path.join(destinationDir, platformFileName),
+        temporaryDir,
+        initialized: true,
+      };
+    }
+
+    return this.archiveData;
+  }
+
+  cleanUp() {
+    if (this.isInitialized()) {
+      this.archiveData = {
+        path: null,
+        temporaryDir: null,
+        initialized: false,
+      };
+    }
+  }
+}
+
+class LocalArchiveProvider extends ArchiveProvider {
+  constructor(path) {
+    super();
+    this.path = path;
+  }
+
+  getType() {
+    return 'local';
+  }
+
+  async getArchivePath() {
+    const data = await this.getArchiveData();
+    return data.path;
+  }
+
+  async getPlatformJsonPath() {
+    const platformJsonFile = fs.readFileSync(path.join(this.path, 'platform', 'platform.json'));
+    return platformJsonFile;
+  }
+
+  async getPlatformJson() {
+    const platformJsonFile = await this.getPlatformJsonPath();
+    const jsonContent = JSON.parse(platformJsonFile);
+
+    return jsonContent;
+  }
+
+  async getArchiveData() {
+    if (!this.isInitialized()) {
+      const temporaryDir = await tmp.dir();
+
+      const destinationDir = temporaryDir.path;
+      const platformFileName = `platform.${ARCHIVE_FORMAT}`;
+
+      const ignores = this.loadIgnoreList();
+
+      const platformJson = await this.getPlatformJson();
+      const rootDirectoryName = `platform-${platformJson.version}`;
+
+      await packItUp(this.path, destinationDir, platformFileName, ignores, rootDirectoryName);
+
+      this.archiveData = {
+        path: path.join(destinationDir, platformFileName),
+        temporaryDir,
+        initialized: true,
+      };
+    }
+
+    return this.archiveData;
+  }
+
+  cleanUp() {
+    if (this.isInitialized()) {
+      this.archiveData = {
+        path: null,
+        temporaryDir: null,
+        initialized: false,
+      };
+    }
+  }
+
+  loadIgnoreList() {
+    const ignoreFilePath = path.join(this.path, SHOUTEM_IGNORE_FILE_NAME);
+    const shoutemIgnores = readLinesInFile(ignoreFilePath);
+
+    if(_.isEmpty(shoutemIgnores)) {
+      console.log("WARNING: missing or empty .shoutemignore file!\nPacking everything into the platform archive...")
+    }
+
+    // no need to remove comments, 'ignores' does that for us
+    return ignore().add(shoutemIgnores);
+  }
+}
+
+export function packItUp(sourcePath, destinationDir, platformFileName, ignores, rootDirectoryName) {
```

Comment:
This needs a better name, perhaps `packDirectory`. Also, the file is named `archive` and is in `services`, suggesting it's generic - while it only concerns platform. Either make it generic, or rename the file and place it in a place more suitable for something that concerns only platform.